osidb/models.py

```diff
@@ -1952,6 +1952,82 @@ def bzsync(self, *args, bz_api_key=None, **kwargs):
         self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
 
 
+class FlawAcknowledgmentManager(ACLMixinManager, TrackingMixinManager):
+    """flaw acknowledgment manager"""
+
+    @staticmethod
+    def create_flawacknowledgment(flaw, name, affiliation, **extra_fields):
+        """return a new flaw acknowledgment or update an existing flaw acknowledgment without saving"""
+        try:
+            flawacknowledgment = FlawAcknowledgment.objects.get(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+            )
+            for attr, value in extra_fields.items():
+                setattr(flawacknowledgment, attr, value)
+            return flawacknowledgment
+
+        except ObjectDoesNotExist:
+            return FlawAcknowledgment(
+                flaw=flaw,
+                name=name,
+                affiliation=affiliation,
+                **extra_fields,
+            )
+
+
+class FlawAcknowledgment(ACLMixin, BugzillaSyncMixin, TrackingMixin):
+    """Model representing flaw acknowledgments"""
+
+    # internal primary key
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    # The name of the person or entity being acknowledged.
+    # max length seen in production as of 02/2023 == 122
+    name = models.CharField(max_length=255)
+
+    # The affiliation of the person being acknowledged.
+    # max length seen in production as of 02/2023 == 86
+    affiliation = models.CharField(max_length=255)
+
+    # Whether this acknowledgment comes from an upstream source.
+    from_upstream = models.BooleanField()
+
+    # one flaw can have many acknowledgments
+    flaw = models.ForeignKey(
+        Flaw, on_delete=models.CASCADE, related_name="acknowledgments"
+    )
+
+    # non operational meta data
+    meta_attr = HStoreField(default=dict)
+
+    objects = FlawAcknowledgmentManager()
+
+    class Meta:
+        """define meta"""
+
+        # A flaw can't have multiple identical acknowledgments.
+        # Can't have identical acknowledgments simultaneously from upstream and not from upstream.
```

Comment:
`not upstream` is probably downstream :)

Reply:
What if it's from an otherwise-unrelated entity? :thinking:

Reply:
Then the line 2011 will still be confusing. I guess that `unique_together` is self-explanatory enough and you can remove that line.

Reply:
OK, removed the comment.